### ISSUE TITLE
[Bug]: Locator/getBy functions work only 1% of the time  

### ISSUE THREAD

**Issue by ClimenteA** [microsoft/playwright#30142]
### Version

1.41.1

### Steps to reproduce

I'm trying to use playwright outside it's test environment.

- new folder
- npm init
- npm i playwright
- playwright install-deps (I've installed all the missing lib* packages on ubuntu)
- package.json - type: module to run the following script:

```js

import playwright from 'playwright';

// node scrapper.js

(async () => {

    console.log("open browser")
    const browser = await playwright.chromium.launch({ headless: false });
    const page = await browser.newPage();

    console.log("go to url")
    await page.goto("https://vuejobs.com/jobs");
    console.log("page loaded")

    console.log("get remote label")
    const elem = page.getByLabel("Remote", { exact: true }).first()

    console.log("elem", elem)
    console.log("click")
    await elem.click()

    console.log("click")

    await page.waitForLoadState("networkidle")

    await browser.close();


})();


```

then in terminal run: `node scrapper.js`

### Expected behavior

The checkbox should be checked.

### Actual behavior

Locator and getBy* funcs work rarely/flaky. 

On click step the timeout error is raised:
```shell
go to url
page loaded
get remote label
elem getByLabel('Remote', { exact: true }).first()
click
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

locator.click: Timeout 30000ms exceeded.
Call log:
  - waiting for getByLabel('Remote', { exact: true }).first()

    at /home/climentea/Documents/rjobs/scrapper.js:20:16 {
  name: 'TimeoutError'
}

Node.js v20.11.1 (same with v18)
```

### Additional context

It might be because I'm not running on Ubuntu LTS version?

### Environment

```shell
System:
    OS: Linux 6.5 Ubuntu 23.10 23.10 (Mantic Minotaur)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
    Memory: 11.79 GB / 15.41 GB
    Container: Yes
  Languages:
    Bash: 5.2.15 - /usr/bin/bash
  npmPackages:
    playwright: ^1.42.1 => 1.42.1
```

**Comment by pavelfeldman** [comment]
I don't see anything matching `const elem = page.getByLabel("Remote", { exact: true }).first()` in this page.

**Comment by ClimenteA** [comment]
True - most of the times I've run the script it didn't find the Remote label tied to the switch (left side). But, in rare tries the script worked.

Maybe page is way to dynamic. I tried various waits with no success.

Same kind lf quirks with this website: https://www.ejobs.ro/locuri-de-munca/remote

I tried getting all the links in page after domcontentloaded and networkidle events and it only got the first 3 listings.

I started creating a chrome extension in order to try to overcome these issues, but got the same result.

Some websites are bundeld with complex js and tools like playwright, pupeeter can't handle them. 

Maybe I'm missing something..

Those 2 websites I think are good for testing playwright limits.

Feel free to close the issue if you want, but I'm very interested if the script provided can be modified to work.
